clvm_tools/cmds.py

```diff
@@ -218,15 +218,16 @@ def launch_tool(args, tool_name, default_stage=0):
     try:
         output = "(didn't finish)"
 
-        use_rust = (
-            (tool_name != "run")
-            and not pre_eval_f
-            and (
-                args.backend == "rust"
-                or (run_chia_program and args.backend != "python")
-            )
-            and args.stage == 0
-        )
+        if args.backend == "rust":
+            use_rust = True
+        elif args.backend == "python":
+            use_rust = False
+        else:
+            use_rust = (tool_name != "run"
+                and not pre_eval_f
+                and run_chia_program
+                and args.stage.__file__.endswith("stage_0.py"))
```

Comment:
`stage` is an imported module from `stages/`

Reply:
How about this one?
```python
args.stage.__name__ == "stages.stage_0"
```